app/test_main.py

```diff
@@ -1 +1,16 @@
-# write your code here
+from typing import Callable
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection_call(mocked: Callable) -> None:
+    can_access_google_page("https://www.google.com/webhp?client=firefox-b-d")
+    mocked.assert_called_once()
+
+
+@mock.patch("app.main.valid_google_url")
+def test_have_valid_google_url(mocked: Callable) -> None:
+    can_access_google_page("https://www.google.com/webhp?client=firefox-b-d")
+    mocked.assert_called_once()
```

Comment:
And here valid_google_url accepts argument so better called assert_called_once_with